### PR TITLE
feat: Check if the certificate has not been revoked before connection

### PR DESCRIPTION
**Pull request by mprevide**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**
Check if the certificate has not been revoked before connection

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)
   dojot:dojot#1201

* **Other information**:

Dependes #75

### CODE REVIEW

File: certUtils.py [hunk 1]
```diff
@@ -12,13 +12,18 @@
 
 def saveCRL(filename, rawCRL):
     crl = ("-----BEGIN X509 CRL-----\n"
-           + re.sub("(.{64})", "\\1\n", rawCRL, 0, re.DOTALL)
+           + re.sub("(.{64})", "\\1\n", rawCRL, 0, re.DOTALL).strip()
```
Comment (mprevide) [inline certUtils.py]: https://stackoverflow.com/questions/2657693/insert-a-newline-character-every-64-characters-using-python
def saveCRL(filename, rawCRL):
crl = ("-----BEGIN X509 CRL-----\n"
+ '\n'.join(rawCRL[i:i+64] for i in range(0, len(rawCRL), 64)) #X509 -each line have 64 bytes
+ "\n-----END X509 CRL-----\n")
crypto.load_crl(crypto.FILETYPE_PEM, crl)

**Comment by muhamedcpqd** [comment]
Very good @mprevide . Nice job!!

File: certUtils.py [hunk 2]
```diff
@@ -9,17 +9,28 @@
 
 # the following functions are related to file manipulation
 # these functions  may throw crypto.Error
+# change to (https://stackoverflow.com/questions/2657693/insert-a-newline-character-every-64-characters-using-python)
```
Comment (muhamedcpqd) [inline certUtils.py]: I think we can use this "pythonic" way to parse the cert data. According with the annexed link, this way is more efficient and less error prone

File: certUtils.py [hunk 3]
```diff
@@ -9,17 +9,28 @@
...
+           + re.sub("(.{64})", "\\1\n", rawCRL, 0, re.DOTALL).strip()
            + "\n-----END X509 CRL-----\n")
+
     crypto.load_crl(crypto.FILETYPE_PEM, crl)
 
     with open(filename, "w") as crlFile:
         crlFile.write(crl)
 
 
+
```
Comment (muhamedcpqd) [inline certUtils.py]: Empty lines.

File: tests/unit-tests/certificates.test.js [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+    });
+
+    it("should build a Cerficiates", () => {
+        require("fs").__createMockFiles(FOLDER_PRESENT_CONFIG);
+        const Certificates = require('../../src/certificates.js');
+        expect(Certificates).toBeDefined();
+        expect(Certificates.getCRLPEM()).toBe(crlNoRevokeRevoke.PEM);
+    });
+
+    it("should build a Cerficiates", () => {
```
Comment (muhamedcpqd) [inline tests/unit-tests/certificates.test.js]: Why three tests with the same purpose? maybe you can merge it into one or say the specific test that the 'it' clause is testing.

**Comment by muhamedcpqd** [comment]
I think its good! If want, you can use my comments to do some changes. If not, i think its ok to merge.

**Comment by mprevide** [comment]
> I think its good! If you want, you can use my comments to do some changes. If not, i think its ok to merge.
I'll use your comments, thx =)

File: tests/unit-tests/app.test.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    });
+
+    it("Should not throw an error", (done) => {
+        // const appInstance = App.app;
+        // expect(appInstance).toBeDefined();
+        done();
+    });
+
+    it("Test initApp", (done) => {
+        // App.initApp = jest.fn(App.initApp);
```
Comment (muhamedcpqd) [inline tests/unit-tests/app.test.js]: Why those tests are commented?

**Pull request closed without merging**